tests/test_config.py

```diff
@@ -0,0 +1,1470 @@
+        ("""\
+operator_manifests:
+  allowed_registries: null
+  repo_replacements:
+    - package_mappings_url: https://somewhere.net/mapping.yaml
+        """, False),  # missing registry
+        ("""\
+operator_manifests:
+  allowed_registries: null,
+  repo_replacements:
+    - registry: foo
+      package_mappings_url: mapping.yaml
+        """, False),  # package mappings url is not a url
+    ])
+    def test_get_operator_manifests(self, tmpdir, config, valid):
+        config += "\n" + REQUIRED_CONFIG
+        if valid:
+            read_yaml(config, 'schemas/config.json')
+        else:
+            with pytest.raises(OsbsValidationException):
+                read_yaml(config, 'schemas/config.json')
+            return
+
+        filename = os.path.join(str(tmpdir), 'config.yaml')
+        with open(filename, 'w') as fp:
+            fp.write(dedent(config))
+        conf = Configuration(config_path=str(tmpdir))
+
+        operator_config = conf.operator_manifests
+        assert isinstance(operator_config, dict)
+        assert "allowed_registries" in operator_config
+
+    def test_set_user_params(self, tmpdir):
+        filename = os.path.join(str(tmpdir), 'config.yaml')
+        with open(filename, 'w') as fp:
+            fp.write(dedent(REQUIRED_CONFIG))
+
+        workflow = self.prepare()
+        Configuration(config_path=str(tmpdir))
+
+        for k, v in USER_PARAMS.items():
+            assert workflow.user_params[k] == v
+
+    @pytest.mark.parametrize('config, valid', [
+        ("""\
+build_env_vars: []
+        """, True),
+        ("""\
+build_env_vars:
+- name: HTTP_PROXY
+  value: example.proxy.net
+- name: NO_PROXY
+  value: localhost
+        """, True),
+        ("""\
+build_env_vars:
+- name: FOO
+  value: 1
+        """, False),  # values must be strings
+        ("""\
+build_env_vars:
+- name: FOO
+        """, False),  # values must be defined
+    ])
+    def test_validate_build_env_vars(self, config, valid):
+        # Only test schema validation, atomic-reactor has no additional support
+        # for build_env_vars (osbs-client does, however)
+        config += "\n" + REQUIRED_CONFIG
+        if valid:
+            read_yaml(config, 'schemas/config.json')
+        else:
+            with pytest.raises(OsbsValidationException):
+                read_yaml(config, 'schemas/config.json')
+
+    @pytest.mark.parametrize(('dockerfile_images', 'organization'), [
+        (True, None),
+        (True, 'organization'),
+        (False, None),
+        (False, 'organization'),
+    ])
+    def test_set_source_registry(self, tmpdir, dockerfile_images, organization):
+        config = REQUIRED_CONFIG
+
+        if organization:
+            config += "\nregistries_organization: " + organization
+
+        filename = os.path.join(str(tmpdir), 'config.yaml')
+        with open(filename, 'w') as fp:
+            fp.write(dedent(config))
+        workflow = self.prepare()
```

Comment:
To me this looks like the only test that actually needs a `workflow` object, could it be removed from the other tests?

Reply:
removed